### PR TITLE
KAA-1580: Kafka log appender - reviewed. [master]

### PR DESCRIPTION
**Pull request by hunch7**

### CODE REVIEW

File: doc/Programming-guide/Key-platform-features/Data-collection/Kafka-log-appender/index.md [hunk 1]
```diff
@@ -151,21 +93,82 @@ Example result:
 }
 ```
 
-# Playing with Kafka log appender
+## Configure log appender
+
+The Cassandra log appender configuration must match [this Avro schema]({{github_url}}server/appenders/kafka-appender/src/main/avro/kafka-appender-config.avsc).
```
Comment (Polinaa) [inline doc/Programming-guide/Key-platform-features/Data-collection/Kafka-log-appender/index.md]: Kafka log appender

File: doc/Programming-guide/Key-platform-features/Data-collection/Kafka-log-appender/index.md [hunk 2]
```diff
@@ -151,21 +93,82 @@ Example result:
...
+|`kafkaServers`           |List of Kafka bootstrap servers (host name and port pairs).                                                              |
+|`partitionCount`         |Count of event partitions.                                                                                             |
+|`retries`                |Failover property. Number of retries for message delivery failure due to no connection.                                             |
+|`topic`                  |Logs destination topic.                                                                                                 |
+|`useDefaultPartitioner`  |If false, the log appender will calculate partitions independently.                                                              |
+
+Below is an example configuration that matches the mentioned Avro schema.
+
+```json
+{
```
Comment (Polinaa) [inline doc/Programming-guide/Key-platform-features/Data-collection/Kafka-log-appender/index.md:118]: Couldn't upload this example configuration in Administration UI: _Unexpected service error occurred: Expected boolean. Got START_OBJECT_
The next modified configuration uploaded successfully:
```
{
    "kafkaServers":[
        {
            "host":"localhost",
            "port":9092
        }
    ],
    "topic":"kaa",
    "useDefaultPartitioner":true,
    "partitionCount":1,
    "kafkaKeyType":{
        "org.kaaproject.kaa.server.appenders.kafka.config.gen.KafkaKeyType":"NULL"
    },
    "executorThreadPoolSize":1,
    "bufferMemorySize":33554432,
    "kafkaCompression":{
        "org.kaaproject.kaa.server.appenders.kafka.config.gen.KafkaCompression":"NONE"
    },
    "kafkaAcknowledgement":{
        "org.kaaproject.kaa.server.appenders.kafka.config.gen.KafkaAcknowledgement":"ONE"
    },
    "retries":0
}
```

File: doc/Programming-guide/Key-platform-features/Data-collection/Kafka-log-appender/index.md [hunk 3]
```diff
@@ -151,21 +93,82 @@ Example result:
...
+
+The example below uses the **Data collection demo** from [Kaa Sandbox]({{root_url}}Glossary/#kaa-sandbox).
+
+Before proceeding with the example below:
+
+1. [Install Kafka](https://kafka.apache.org/07/quickstart.html).
+
+2. Change Zookeeper to use some other port than 2181 (e.g. 2183).
+
+3. Edit the following property files:
```
Comment (Polinaa) [inline doc/Programming-guide/Key-platform-features/Data-collection/Kafka-log-appender/index.md]: This step relates to the previous.

File: doc/Programming-guide/Key-platform-features/Data-collection/Kafka-log-appender/index.md [hunk 4]
```diff
@@ -196,58 +199,41 @@ The following JSON example matches the schema above.
...
-    {"header":{"endpointKeyHash":{"string":"tqoeo8S49HgakOV/2DfiEZLjGls="},"applicationToken":{"string":"24212667430286144698"},"headerVersion":{"int":1},"timestamp":{"long":1474622333932},"logSchemaVersion":{"int":2}},"event":{"temperature":28,"timeStamp":1474622330}}
-    {"header":{"endpointKeyHash":{"string":"tqoeo8S49HgakOV/2DfiEZLjGls="},"applicationToken":{"string":"24212667430286144698"},"headerVersion":{"int":1},"timestamp":{"long":1474622333932},"logSchemaVersion":{"int":2}},"event":{"temperature":28,"timeStamp":1474622330}}
-    {"header":{"endpointKeyHash":{"string":"tqoeo8S49HgakOV/2DfiEZLjGls="},"applicationToken":{"string":"24212667430286144698"},"headerVersion":{"int":1},"timestamp":{"long":1474622333932},"logSchemaVersion":{"int":2}},"event":{"temperature":28,"timeStamp":1474622330}}
-    ```
-
-If your output doesn't match above one, please follow our [troubleshooting guide]({{root_url}}Administration-guide/Troubleshooting).
+To play around with the Kafka log appender:
+
+1. Open the Data collection demo from Kaa Sandbox, follow the application installation instructions, and run the application.
```
Comment (Polinaa) [inline doc/Programming-guide/Key-platform-features/Data-collection/Kafka-log-appender/index.md]: You don't need to run the application in this step.

**Review by Polinaa — APPROVED** [review]

File: doc/Programming-guide/Key-platform-features/Data-collection/Kafka-log-appender/index.md (hunk 1 above)
Comment (hunch7) [reply doc/Programming-guide/Key-platform-features/Data-collection/Kafka-log-appender/index.md]: Fixed.

File: doc/Programming-guide/Key-platform-features/Data-collection/Kafka-log-appender/index.md (hunk 2 above)
Comment (hunch7) [reply doc/Programming-guide/Key-platform-features/Data-collection/Kafka-log-appender/index.md:118]: Fixed.

File: doc/Programming-guide/Key-platform-features/Data-collection/Kafka-log-appender/index.md (hunk 4 above)
Comment (hunch7) [reply doc/Programming-guide/Key-platform-features/Data-collection/Kafka-log-appender/index.md]: Fixed.

File: doc/Programming-guide/Key-platform-features/Data-collection/Kafka-log-appender/index.md (hunk 3 above)
Comment (hunch7) [reply doc/Programming-guide/Key-platform-features/Data-collection/Kafka-log-appender/index.md]: Fixed.

**Review by Polinaa — APPROVED** [review]

**Review by vadimol — APPROVED** [review]

**Pull request merged**